src/cmd/manage_nsfs.js

```diff
@@ -0,0 +1,411 @@
+        if (!config_root) {
+            console.error('Error: Config dir should not be empty');
+            print_account_usage();
+            return;
+        }
+        const state = await check_and_create_config_dirs(config_root);
+        if (state) {
+            console.log("NSFS config root dirs are created in :" + config_root);
+        }
+        if (resourse_type === 'account') {
+           account_management(argv);
+        }
+        if (resourse_type === 'bucket') {
+            bucket_management(argv);
+        }
+    } catch (err) {
+      console.error('NSFS Manage command: exit on error', err.stack || err);
+      process.exit(2);
+    }
+}
+
+async function bucket_management(argv) {
+    const config_root = argv.config_root ? String(argv.config_root) : '';
+    const action = argv._[1] || '';
+    const data = await fetch_bucket_data(argv);
+    manage_bucket_operations(action, data, config_root);
+}
+
+async function fetch_bucket_data(argv) {
+    const action = argv._[1] || '';
+    const config_root = argv.config_root ? String(argv.config_root) : '';
+    const name = argv.name && new SensitiveString(String(argv.name));
+    const email = argv.email && new SensitiveString(String(argv.email));
+    const from_file = argv.from_file ? String(argv.from_file) : '';
+    let data;
+    if (from_file) {
+        const raw_data = await fs.promises.readFile(String(argv.from_file));
+        data = JSON.parse(raw_data.toString());
+        data.name = new SensitiveString(data.name);
+    }
+    if (!data) {
+        const bucket_path = argv.path ? String(argv.path) : '';
+        data = {
+            name: name,
+            system_owner: email,
+            bucket_owner: email,
+            tag: "",
+            versioning: "DISABLED",
+            path: bucket_path,
+            should_create_underlying_storage: true,
+        };
+    }
+    if (action === 'update') {
+        data = _.omitBy(data, _.isUndefined);
+        data = update_bucket(config_root, data);
+    }
+    return data;
+}
+
+async function update_bucket(config_root, target) {
+    const account_path = config_root + '/buckets';
+    let source;
+    try {
+        const source_raw = await fs.promises.readFile(path.join(account_path, target.name + '.json'));
+        source = JSON.parse(source_raw.toString());
+        //target.nsfs_account_config.new_buckets_path = target.nsfs_account_config.new_buckets_path || source.nsfs_account_config.new_buckets_path;
+    } catch (err) {
+        console.error('NSFS Manage command: Could not find bucket ' + target.name + ' to update');
+        print_account_usage();
+    }
+    const data = _.merge({}, source, target);
+    return data;
+}
+
+async function manage_bucket_operations(action, data, config_root) {
+    const bucket_config_path = config_root + '/buckets';
+    if (action === 'add') {
+        const is_valid = await validate_bucket_add_args(data);
```

Comment:
I would suggest to split this function to multiple functions, each handling an action